cdc/puller/ddl_puller.go

```diff
@@ -42,6 +49,154 @@ const (
 	DDLPullerTableName = "DDL_PULLER"
 )
 
+// DDLJobPuller is used to pull ddl job from TiKV.
+type DDLJobPuller interface {
+	Run(ctx context.Context) error
+	Output() <-chan *model.DDLJobEntry
+}
+
+type ddlJobPullerImpl struct {
+	puller    Puller
+	kvStorage tidbkv.Storage
+	tableInfo *model.TableInfo
+	columnID  int64
+	outputCh  chan *model.DDLJobEntry
+}
+
+func (p *ddlJobPullerImpl) Run(ctx context.Context) error {
+	g, ctx := errgroup.WithContext(ctx)
+
+	g.Go(func() error {
+		return p.puller.Run(ctx)
+	})
+
+	rawDDLCh := memory.SortOutput(ctx, p.puller.Output())
+	var ddlRawKV *model.RawKVEntry
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case ddlRawKV = <-rawDDLCh:
+		}
+		if ddlRawKV == nil {
+			continue
+		}
+		job, err := p.unmarshalDDL(ddlRawKV)
+		p.outputCh <- &model.DDLJobEntry{
```

Comment:
Please select on `p.outputCh <- ...` and `<-ctx.Done()`, or otherwise it's very likely to cause a deadlock problem.